kentikapi/client.go

```diff
@@ -0,0 +1,75 @@
+package kentikapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+)
+
+// Public constants.
+const (
+	APIURLUS = "https://api.kentik.com/api/v5"
+	APIURLEU = "https://api.kentik.eu/api/v5"
+)
+
+// Client is Kentik API client.
+type Client struct {
+	config     ClientConfig
+	restClient *restClient
+}
+
+// ClientConfig holds configuration of the client.
+type ClientConfig struct {
+	// APIURL defaults to "https://api.kentik.com/api/v5"
+	APIURL    string
+	AuthEmail string
+	AuthToken string
+}
+
+// NewClient creates a new client.
+func NewClient(c ClientConfig) *Client {
+	if c.APIURL == "" {
+		c.APIURL = APIURLUS
+	}
+
+	return &Client{
+		config: c,
+		restClient: newRestClient(restClientConfig{
+			APIURL:    c.APIURL,
+			AuthEmail: c.AuthEmail,
+			AuthToken: c.AuthToken,
+		}),
+	}
+}
+
+// GetAllUsers lists users.
+func (c *Client) GetAllUsers(ctx context.Context) (_ []models.User, err error) {
```

Comment:
I modified the code and the Client has `UsersAPI usersAPI` field now. It achieves all the points you listed, I think.

There more possible refactorings:
- expose Client.UsersAPI interface instead of Client.usersAPI struct.
- move groups of APIs to a subpackage, e.g. `kentikapi/apiresources` (like in Py lib)

I am not sure it is worth to do so, tough.